src/footer/script1.js

```diff
@@ -0,0 +1,222 @@
+if (isCart() == true) {
+    getResults();
+    setInterval(getResults, 2000);
+}
+
+async function getRecommnededProductsFromServer(us_cart_items) {
+    if (shouldCallServer(us_cart_items) === false) {
+        return null;
+    }
+    const us_project_id = getShoptetDataLayer('projectId');
+    const us_project_id_modulo = us_project_id % 11;
+    try {
+        const response = await fetch('https://slabihoud.cz/products/' + us_project_id + '/' + us_project_id_modulo + '/' + us_cart_items.toString());
+        const data = await response.json();
+        return data;
+    } catch (error) {
+        console.error('Error:', error);
+    }
+}
+
+function shouldCallServer(us_cart_items) {
+    if (us_cart_items.length == 0) {
+        return false;
+    }
+
+    if (sessionStorage.getItem('us_request_' + us_cart_items.toString()) != null) {
+        return false;
+    }
+    return true;
+}
+
+function getCartItemsGUIDS() {
+    let us_result_items = [];
+    const us_cart_items = getCartItems();
+    us_cart_items.forEach(function (el) {
+        us_result_items.push(el.getAttribute('data-micro-identifier'));
+    });
+    us_result_items.sort();
+    return us_result_items;
+}
+
+function getCartItems() {
+    return document.querySelectorAll('[data-micro-identifier]:not([data-source="easy-upsell"])');
+}
+
+function getImageCdn() {
+    const us_cart_items = document.querySelectorAll('td.cart-p-image a img');
+    if (us_cart_items.length > 0) {
+        let us_image_src = us_cart_items[0].getAttribute('data-src');
+        const us_image_parts = us_image_src.split('/');
+        us_image_parts.pop();
+        const us_image_cdn = us_image_parts.join('/');
+        return us_image_cdn + '/';
+    }
+}
+
+function isCart() {
+    if (getShoptetDataLayer('pageType') == 'cart') {
+        return true;
+    }
+    return false;
+}
+
+async function getResults() {
+    setMainDiv();
+    const us_cart_items = getCartItemsGUIDS();
+    const result = await getRecommnededProductsFromServer(us_cart_items);
+    if (result != null) {
+        await cacheResults(result);
+        await cacheRequest(us_cart_items, result);
+    }
+    printResults();
+}
+
+function setMainDiv() {
+    const us_main_div = document.querySelectorAll('#upsell-container');
+    if (us_main_div.length == 0) {
+        const us_cart = document.querySelectorAll('.cart-table');
+        if (us_cart.length > 0) {
+            us_cart.forEach(async function (el) {
+                el.insertAdjacentHTML('afterend', '<hr><div id="upsell-container"></div>');
+            });
+        }
+    } else {
+        us_main_div.forEach(function (el) {
+            el.innerHTML = '';
+        });
+    }
+}
+
+async function cacheResults(result) {
+    result.recommendations.forEach(async function (recommendation) {
+        recommendation = await getVariantDetail(recommendation);
```

Comment:
Nelze toto nahradit rovnou a zbavit se funkce `getVariantDetail`:
```suggestion
        recommendation = await getVariantDetailFromEshop(recommendation);
```